stl/inc/format

```diff
@@ -2793,6 +2843,24 @@ template <_Format_supported_charT _CharT, class _Traits>
 struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
     : _Formatter_base<basic_string_view<_CharT, _Traits>, _CharT, _Basic_format_arg_type::_String_type> {};
 
+template <class _CharT, class... _Args>
+struct _Basic_format_string {
+    basic_string_view<_CharT> _Str;
+
+    template <convertible_to<basic_string_view<_CharT>> _Ty>
+    consteval _Basic_format_string(const _Ty& _Str_val) : _Str(_Str_val) {

```

Comment:
Could we work around the `consteval` issues by making this `constexpr` and adding an `if (!_STD is_constant_evaluated())`

Reply:
not really, since that doesn't comply with the standard. That means that strings that aren't constants will just skip the checks, which isn't what the standard says to do.

We've fixed the relevant consteval bug and will backport said fix along with this feature (I think it might already be backported)